interfaces/Portal/src/app/services/language.service.ts

```diff
@@ -0,0 +1,96 @@
+import { Injectable } from '@angular/core';
+import { TranslateService } from '@ngx-translate/core';
+import { environment } from '../../environments/environment';
+
+export class LanguageOption {
+  key: string;
+  name: string;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LanguageService {
+  private DEFAULT_LANGUAGE_KEY = 'en';
+  private DEFAULT_LANGUAGE_LABEL = 'ENGLISH';
+
+  private selectedLanguage: string;
+
+  constructor(private translate: TranslateService) {
+    if (!this.translate) {
+      return;
+    }
+    this.translate.setDefaultLang(this.DEFAULT_LANGUAGE_KEY);
+
+    if (!this.selectedLanguage && !this.getStoredLanguage()) {
+      this.selectedLanguage = this.DEFAULT_LANGUAGE_KEY;
+      this.storeLanguage();
+    }
+
+    this.selectedLanguage = this.getStoredLanguage();
+
+    this.translate.use(this.selectedLanguage);
```

Comment:
I think that setting the document Language + direction needs to happen right after here... (and not in the current place)

```suggestion
    this.translate.use(this.selectedLanguage);
    document.documentElement.lang = this.selectedLanguage;
    document.documentElement.dir = this.translate.instant('_dir');
```

Or is the `lang`-attribute already set by the ngx-translate-package?